File: 0066-plus-one/0066-plus-one.java
class Solution {
    public int[] plusOne(int[] digits) {
        int add = 1;
        int mod = 0;
        for (int i = digits.length - 1; i >= 0; i --){
            mod = (digits[i] + add) % 10;
            add = ((int) ((digits[i] + add) / 10));
            digits[i] = mod;
        }
        if (add > 0) {
            int[] newDigits = new int[digits.length + 1];
            newDigits[0] = add;
            System.arraycopy(digits, 0, newDigits, 1, digits.length);
            digits = newDigits;
            // digits.append(add);
        }
        return digits;
    }
}

File: 0066-plus-one/0066-plus-one.py
class Solution(object):
    def plusOne(self, digits):
        """
        :type digits: List[int]
        :rtype: List[int]
        """
        if 9 == digits[-1]:
            s = 0
            for i, dig in enumerate(digits[::-1]):
                s += dig*10**i
            return [int(d) for d in str(s + 1)]
        else:
            return digits[:len(digits[:-1])] + [digits[-1] + 1]
            # return digits[-1] + 1